src/widgets/table.rs

```diff
@@ -388,136 +453,428 @@ impl<'a> Table<'a> {
+    ///
+    /// ```rust
+    /// # use ratatui::{prelude::*, widgets::*};
+    /// # let rows = vec![Row::new(vec!["Cell1", "Cell2"])];
+    /// let table = Table::new(rows).style(Style::new().red().italic());
+    /// ```
+    ///
+    /// `Table` also implements the [`Styled`] trait, which means you can use style shorthands from
+    /// the [`Stylize`] trait to set the style of the widget more concisely.
+    ///
+    /// ```rust
+    /// # use ratatui::{prelude::*, widgets::*};
+    /// # let rows = vec![Row::new(vec!["Cell1", "Cell2"])];
+    /// let table = Table::new(rows).red().italic();
+    /// ```
+    pub fn style(mut self, style: Style) -> Self {
+        self.style = style;
+        self
+    }
+
+    /// Set the style of the selected row
+    ///
+    /// This style will be applied to the entire row, including the selection symbol if it is
+    /// displayed, and will override any style set on the row or on the individual cells.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use ratatui::{prelude::*, widgets::*};
+    /// # let rows = vec![Row::new(vec!["Cell1", "Cell2"])];
+    /// let table = Table::new(rows).highlight_style(Style::new().red().italic());
+    /// ```
     pub fn highlight_style(mut self, highlight_style: Style) -> Self {
         self.highlight_style = highlight_style;
         self
     }
 
+    /// Set the symbol to be displayed in front of the selected row
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use ratatui::{prelude::*, widgets::*};
+    /// # let rows = vec![Row::new(vec!["Cell1", "Cell2"])];
+    /// let table = Table::new(rows).highlight_symbol(">>");
+    /// ```
+    pub fn highlight_symbol(mut self, highlight_symbol: &'a str) -> Self {
+        self.highlight_symbol = Some(highlight_symbol);
+        self
+    }
+
     /// Set when to show the highlight spacing
     ///
-    /// See [`HighlightSpacing`] about which variant affects spacing in which way
+    /// The highlight spacing is the spacing that is allocated for the selection symbol column (if
+    /// enabled) and is used to shift the table when a row is selected. This method allows you to
+    /// configure when this spacing is allocated.
+    ///
+    /// - `HighlightSpacing::Always` will always allocate the spacing, regardless of whether a row
+    ///   is selected or not. This means that the table will never change size, regardless of if a
+    ///   row is selected or not.
+    /// - `HighlightSpacing::WhenSelected` will only allocate the spacing if a row is selected. This
+    ///   means that the table will shift when a row is selected. This is the default setting for
+    ///   backwards compatibility, but it is recommended to use `HighlightSpacing::Always` for a
+    ///   better user experience.
+    /// - `HighlightSpacing::Never` will never allocate the spacing, regardless of whether a row is
+    ///   selected or not. This means that the highlight symbol will never be drawn.
```

Comment:
This one I'm fine with leaving duplicated, because it gives you the information you need when you're tab completing a method or hovering over an existing one, so at that point you don't have an enum to get docs on. On the flipside the enum may end up in some variable or as a paramter, so the info is useful there too.

Reply:
Yep this is reasonable